### PR TITLE
The basic enrollment models.

### PR DESCRIPTION
**Pull request by iloveagent57**
## Description

TODO: Include a detailed description of the changes in the body of the PR

### CODE REVIEW

File: registrar/apps/enrollments/models.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+
+class Program(TimeStampedModel):
+    """
+    Table that referrences a course-discovery Program entity.
+    """
+    class Meta(object):
+        app_label = 'enrollments'
+
+    catalog_uuid = models.UUIDField(db_index=True, null=True)
```
Comment (schenedx) [inline registrar/apps/enrollments/models.py]: Should we rename this to discovery_uuid to not confuse it with the actual course catalog concept?

File: registrar/apps/enrollments/models.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    """
+    organization = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    course_run = models.ForeignKey(CourseRun, on_delete=models.CASCADE)
+
+
+class CourseRunEnrollmentStatus(TimeStampedModel):
+    """
+    Model that represents possible CourseRun enrollment statuses.
+    """
+    key = models.CharField(max_length=64, db_index=True, null=False)
```
Comment (schenedx) [inline registrar/apps/enrollments/models.py]: How would we leverage the status here to drive other logic within code if these are defined during run time? For example, we want to update the LearnerCourseRunEnrollment later from "Pending" to "active". These are values defined by systems, not by users, right?

**Review by schenedx — COMMENTED** [review]
See my inline comments. I like how thorough this is.

File: registrar/apps/enrollments/models.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    title = models.CharField(max_length=512)
+    key = models.CharField(max_length=255, db_index=True)
+    organizations = models.ManyToManyField(
+        Organization,
+        related_name='course_runs',
+        through='OrganizationCourseRunMembership'
+    )
+
+
+class OrganizationCourseRunMembership(TimeStampedModel):
```
[the remark this answers is not in the document]
Comment (iloveagent57) [reply registrar/apps/enrollments/models.py]: I'm thinking about programs and course runs being associated with orgs just to check that a client of this app is allowed to enroll students in the programs and courses being requested.  I don't want someone from GT to accidentally enroll students in a program that they're not involved with.

File: registrar/apps/enrollments/models.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    program = models.ForeignKey(Program, on_delete=models.PROTECT)
+    status = models.ForeignKey(ProgramEnrollmentStatus, on_delete=models.PROTECT)
+    history = HistoricalRecords()
+
+
+class CourseRun(TimeStampedModel):
+    """
+    Model that represents a course-discovery CourseRun entity.
+    """
+    catalog_uuid = models.UUIDField(db_index=True, null=True)
```
[the remark this answers is not in the document]
Comment (iloveagent57) [reply registrar/apps/enrollments/models.py]: sure.

File: registrar/apps/enrollments/models.py (hunk 1 above)
Comment (iloveagent57) [reply registrar/apps/enrollments/models.py]: sure.

File: registrar/apps/enrollments/models.py (hunk 2 above)
Comment (iloveagent57) [reply registrar/apps/enrollments/models.py]: I was thinking of hard-cording some enums in the code, and they would also be persisted.  Kind of like we do for course modes.

Comment (schenedx) [reply registrar/apps/enrollments/models.py]: I see, you just haven't write down what those ENUMs are. OK. I got ya

File: registrar/apps/enrollments/models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (schenedx) [reply registrar/apps/enrollments/models.py]: Hmm... sounds like a validation we don't need at the moment? Maybe remove it for now. And if we proved that we need it later, we can always add that. Let's not have this model for now if we don't see a clear use case for it?

**Pull request merged**